### PR TITLE
Brandi Kinard patch 1 - Initial attempt to introduce ML to the project

### PR DESCRIPTION
**Pull request by Brandi-Kinard**
In incremental steps, this was my first attempt at modifying my Flask project to introduce a clustering-based recommendation system for additional recipe suggestions. This involved both modifications to my existing Python code and the introduction of new logic for clustering and recommendation. 

image

### CODE REVIEW

**Comment by Brandi-Kinard** [comment]
TL;DR
Based on the screenshot, this was a failed attempt. The code should not be grouping (clustering) to find the best recipe and suggest additional recipes. If this is the case, I will modify the code to have only the additional suggested recipes utilize clustering. 

---

As you can see in the screenshot I attached above, this attempt was not successful -- I didn't get a best match recipe based on my ingredient selections, instead I received a "No matching recipes found. Please try different ingredients." message. This is incorrect as I expected the "Peanut Butter" recipe, considering it contains most of my selected ingredients. Additionally, there were no additional suggested recipes. The "Avocado-Coconut" recipes could be suggested due to the 'avocado' and 'mango' ingredients possibly being a common combination based on clustering. I will need to create another branch and revise the code to reflect my actual intentions.

**Comment by Brandi-Kinard** [comment]
Voila! My code is up and running after squashing a tiny bug in my 'app.py' file. I mean, who doesn't love a good bug hunt, right? 🐞

But hold your horses, the coding ride isn't over yet. After playing around with a few ingredient combos, I noticed something odd. When I select 'yogurt' and 'mangos', I get... nada. Zip. Zilch. Not even my friendly "No matching recipes found. Please try different ingredients." message. 🤔

Despite this little hiccup, I'm pretty stoked with how things are shaping up. So, I'm going to merge this branch into the main. The adventure continues in the next branch where I'll be diving back into the code. Catch you on the flip side! ✌️

image
